src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt

```diff
@@ -0,0 +1,58 @@
+package com.hibob.academy.dao
+
+import com.hibob.academy.utils.BobDbTest
+import jakarta.inject.Inject
+import org.jooq.DSLContext
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.Assertions.assertEquals
+import kotlin.random.Random
+import java.time.LocalDate
+
+@BobDbTest
+class PetDaoTest @Inject constructor(private val sql: DSLContext) {
+
+    private val table = PetTable.petInstance
+    private val companyId = Random.nextLong()
+    private val petDao = PetDao(sql)
+
+    @Test
+    fun `insert pet test`() {
+
+        petDao.insertPet(name = "Rex", type = PetTypes.Dog.toString(), companyId = companyId, dateOfArrival = LocalDate.of(2024,1, 1))
+        petDao.insertPet(name = "Rexi", type = PetTypes.Dog.toString(), companyId = companyId, dateOfArrival = LocalDate.of(2024,1, 2))
+
+        val fetchedPets = petDao.getPetsByType(PetTypes.Dog, companyId)
+
+        assertEquals(2, fetchedPets.size)
```

Comment:
Why not test the data class that returned? the size really tells me anything?

Reply:
changed